src/Shared/VolumeNameGenerator.cs

```diff
@@ -20,8 +20,10 @@ public static string CreateVolumeName<T>(IResourceBuilder<T> builder, string suf
             throw new ArgumentException($"The suffix '{suffix}' contains invalid characters. Only [a-zA-Z0-9_.-] are allowed.", nameof(suffix));
         }
 
-        // Create volume name like "myapplication-postgres-data"
-        var applicationName = builder.ApplicationBuilder.Environment.ApplicationName;
+        // Create volume name like "{sha256}-postgres-data"
+
+        // Compute a short hash of the content root path to differentiate between multiple AppHost projects with similar volume names
+        var applicationName = builder.ApplicationBuilder.Configuration["AppHost:Sha256"]![..10].ToLowerInvariant();
```

Comment:
Who sets `AppHost:Sha256`? Do we need to be resilent against `AppHost:Sha256` not having a value? Or someone setting a value shorter than 10?

Reply:
We do: https://github.com/dotnet/aspire/blob/4d80745955ccc0c9b31b1a8fe94e8d5b5b8a1d77/src/Aspire.Hosting/DistributedApplicationBuilder.cs#L154-L169

Nevermind.